src/sd.jl

```diff
@@ -415,3 +415,158 @@ function build_constraint(
+ real(H[1,2]) + (-0.0 - 1.0im) imag(H[1,2])  real(H[2,2])                                real(H[2,3]) + (0.0 + 1.0im) imag(H[2,3])
+ real(H[1,3]) + (-0.0 - 1.0im) imag(H[1,3])  real(H[2,3]) + (-0.0 - 1.0im) imag(H[2,3])  real(H[3,3])
+
+ julia> v = all_variables(model)
+ 9-element Vector{VariableRef}:
+  real(H[1,1])
+  real(H[1,2])
+  real(H[2,2])
+  real(H[1,3])
+  real(H[2,3])
+  real(H[3,3])
+  imag(H[1,2])
+  imag(H[1,3])
+  imag(H[2,3])
+
+julia> all_constraints(model, Vector{VariableRef}, MOI.HermitianPositiveSemidefiniteConeTriangle)
+1-element Vector{ConstraintRef{Model, MathOptInterface.ConstraintIndex{MathOptInterface.VectorOfVariables, MathOptInterface.HermitianPositiveSemidefiniteConeTriangle}}}:
+ [real(H[1,1]), real(H[1,2]), real(H[2,2]), real(H[1,3]), real(H[2,3]), real(H[3,3]), imag(H[1,2]), imag(H[1,3]), imag(H[2,3])] in MathOptInterface.HermitianPositiveSemidefiniteConeTriangle(3)
+```
+We see in the output of the last commands that 9 real variables were created.
+The matrix `H` contrains affine expressions in terms of these 9 variables that
+parametrize a Hermitian matrix.
+"""
+struct HermitianPSDCone end
+
+"""
+    HermitianMatrixShape
+
+Shape object for a Hermitian square matrix of `side_dimension` rows and
+columns. The vectorized form corresponds to
+[`MOI.HermitianPositiveSemidefiniteConeTriangle`](@ref).
+"""
+struct HermitianMatrixShape <: AbstractShape
+    side_dimension::Int
+end
+
+function vectorize(matrix::Matrix, ::HermitianMatrixShape)
+    n = LinearAlgebra.checksquare(matrix)
+    return vcat(
+        vectorize(_real.(matrix), SymmetricMatrixShape(n)),
+        vectorize(
+            _imag.(matrix[1:(end-1), 2:end]),
+            SymmetricMatrixShape(n - 1),
+        ),
+    )
+end
+
+function reshape_vector(v::Vector{T}, shape::HermitianMatrixShape) where {T}
+    NewType = _MA.promote_operation(_MA.add_mul, T, Complex{Bool}, T)
+    n = shape.side_dimension
+    matrix = Matrix{NewType}(undef, n, n)
+    real_k = 0
+    imag_k = MOI.dimension(MOI.PositiveSemidefiniteConeTriangle(n))
+    for j in 1:n
+        for i in 1:(j-1)
+            real_k += 1
+            imag_k += 1
+            matrix[i, j] = v[real_k] + im * v[imag_k]
+            matrix[j, i] = v[real_k] - im * v[imag_k]
+        end
+        real_k += 1
+        matrix[j, j] = v[real_k]
+    end
+    return matrix
+end
+
+function _mapinfo(f::Function, v::JuMP.ScalarVariable)
+    info = v.info
+    return ScalarVariable(
+        VariableInfo(
+            info.has_lb,
+            f(info.lower_bound),
+            info.has_ub,
+            f(info.upper_bound),
+            info.has_fix,
+            f(info.fixed_value),
+            info.has_start,
+            f(info.start),
+            info.binary,
+            info.integer,
+        ),
+    )
+end
+
+_real(s::String) = string("real(", s, ")")
+_imag(s::String) = string("imag(", s, ")")
```

Comment:
If the variable is anonymous, then the names will become `real()` and `imag()`. I'm not sure if `_[1]` or `real(_[1])` is better.